README.md
# tw_step_counter

This is a simple step caculator widget.

pub：[tw_step_counter](https://pub.flutter-io.cn/packages/tw_step_counter)

## introduce

![](https://github.com/zeqinjie/tw_step_counter/blob/main/assets/1.gif)


## Installing

Add tw_step_counter to your pubspec.yaml file:

```yaml
dependencies:
  tw_step_counter: latest_version
```

import tw_step_counter in files that it will be used:
```dart
import 'package:tw_step_counter/tw_step_counter.dart';
```

## Getting Started
some support property 
```dart
/// 每次按钮点击递增递减值
  final double differValue;

  /// 每次输入倍数，失去焦点时触发
  final double? inputMultipleValue;

  /// 支持最小值
  final double mixValue;

  /// 支持最大值
  final double maxValue;

  /// 按钮宽度
  final double? btnWidth;

  /// 高度
  final double? height;

  /// 值颜色
  final Color? valueColor;

  /// 值字体大小
  final double? valueFontSize;

  /// 单位
  final String? unit;

  /// 单位颜色
  final Color? unitColor;

  /// 单位字体大小
  final double? unitFontSize;

  /// 当前值
  final double? currentValue;

  /// 点击回调
  final void Function(double value)? onTap;

  /// 输入的回调
  final void Function(double value)? inputTap;

  /// 默认颜色
  final Color? iconColor;

  /// 加按钮图标，默认系统
  final String? addIcon;

  /// 减按钮图标，默认系统
  final String? reduceIcon;

  /// 禁止加按钮图标，默认系统
  final String? addForbiddenIcon;

  /// 禁止减按钮图标，默认系统
  final String? reduceForbiddenIcon;

  /// 禁止点击颜色
  final Color? forbiddenIconColor;

  /// 保留多少位小数位
  final int decimalsCount;

  /// 高亮颜色
  final Color? highlightColor;

  /// 默认背景色
  final Color? defaultColor;

  /// 边线颜色
  final Color? borderLineColor;

  /// 间隙
  final EdgeInsetsGeometry? padding;

  /// 值的组件间隙
  final EdgeInsetsGeometry? valuePadding;

  /// 控制器
  final TWStepCounterController? controller;

  /// 是否自动限制值范围，默认会
  final bool isUpdateInLimitValue;

  /// 是否支持动画，默认会
  final bool isSupportAnimation;

  /// 是否输入时候更新
  final bool isUpdateInputChange;

  /// 限制输入的长度
  final int? limitLength;
  
```
example

```dart
TWStepCounter(
        unit: '元/天',
        currentValue: 130,
        mixValue: 75,
        maxValue: 250,
        onTap: (value) {
          print('点击回调===>$value');
        },
        inputTap: (value) {
          print('输入回调===>$value');
        },
        controller: controller,
        addForbiddenIcon: 'assets/forbidden_add.png',
        addIcon: 'assets/add.png',
        reduceForbiddenIcon: 'assets/forbidden_reduce.png',
        reduceIcon: 'assets/reduce.png',
        defaultColor: TWColors.twF5F5F5,
        highlightColor: Colors.yellow,
        borderLineColor: Colors.red,
        inputMultipleValue: 5,
        // limitLength: 8,
        // height: 70,
        // isUpdateInLimitValue: false,
        // isSupportAnimation: false,
        decimalsCount: 2,
        btnWidth: 64,
        height: 30,
        isUpdateInputChange: false,
        valuePadding: const EdgeInsets.only(
          left: 10,
          right: 10,
        ),
      )
```



CHANGELOG.md
<!--
 * @Author: zhengzeqin
 * @Date: 2022-07-18 11:14:42
 * @LastEditTime: 2022-08-03 16:50:01
 * @Description: your project
-->

## 0.1.7
- support custom icon

## 0.1.6
- support for control updates on input
- fix bug

## 0.1.5
- support limit length
## 0.1.4
- support input multiple value

## 0.1.3
- fix input animation bug
- fix the input decimal bug
## 0.1.2
- support focus controller
- support input text and range limit
- todo: fix the input decimal bug

## 0.1.1
- fix bug

## 0.1.0

* simple step caculator
- support animation

## 0.0.1

* Initial release


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:tw_step_counter_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(const MyApp());

    // Verify that platform version is retrieved.
    expect(
      find.byWidgetPredicate(
        (Widget widget) => widget is Text &&
                           widget.data!.startsWith('Running on:'),
      ),
      findsOneWidget,
    );
  });
}


test/tw_step_counter_test.dart
/*
 * @Author: zhengzeqin
 * @Date: 2022-07-17 22:49:38
 * @LastEditTime: 2022-07-24 21:19:59
 * @Description: your project
 */

void main() {
  // const MethodChannel channel = MethodChannel('tw_step_counter');

  // TestWidgetsFlutterBinding.ensureInitialized();

  // setUp(() {
  //   channel.setMockMethodCallHandler((MethodCall methodCall) async {
  //     return '42';
  //   });
  // });

  // tearDown(() {
  //   channel.setMockMethodCallHandler(null);
  // });
}


lib/tw_colors.dart
import 'package:flutter/material.dart';

class TWColors {
  TWColors._();

  static const Color tw282828 = Color(0XFF282828);
  static const Color twF5F5F5 = Color(0XFFF5F5F5);
  static const Color tw969696 = Color(0XFF969696);
  static const Color twFF6600 = Color(0XFFFF6600);
  static const Color tw0D74C6 = Color(0XFF0D74C6);
  static const Color twEAEAEA = Color(0XFFEAEAEA);
  static const Color tw999999 = Color(0XFF999999);
  static const Color twA6A6A6 = Color(0XFFA6A6A6);
  static const Color twEFEFEF = Color(0XFFEFEFEF);
  static const Color twF8F8F8 = Color(0XFFF8F8F8);
  static const Color twFFFFFF = Color(0XFFFFFFFF);
  static const Color twE5E5E5 = Color(0XFFE5E5E5);
  static const Color twE7E7E7 = Color(0XFFE7E7E7);
  static const Color twDCDCDC = Color(0XFFDCDCDC);
  static const Color tw868686 = Color(0XFF868686);
  static const Color twFF8000 = Color(0XFFFF8000);
  static const Color tw646464 = Color(0XFF646464);
  static const Color tw04844D = Color(0XFF04844D);
  static const Color twF1F1F1 = Color(0XFFF1F1F1);
  static const Color tw1F1F1F = Color(0XFF1F1F1F);
  static const Color twF74C30 = Color(0XFFF74C30);
  static const Color twFF4200 = Color(0XFFFF4200);
  static const Color twFF4000 = Color(0XFFFF4000);
  static const Color twFF0000 = Color(0XFFFF0000);
  static const Color twFE8813 = Color(0XFFFE8813);
  static const Color twFE9956 = Color(0XFFFE9956);
  static const Color tw29B75B = Color(0XFF29B75B);
  static const Color twC4C4C4 = Color(0XFFC4C4C4);
  static const Color twF4F4F4 = Color(0XFFF4F4F4);
  static const Color twD8D8D8 = Color(0XFFD8D8D8);
  static const Color twBBBBBB = Color(0XFFBBBBBB);
  static const Color twFFB56C = Color(0XFFFFB56C);
  static const Color tw4C4C4C = Color(0XFF4C4C4C);
  static const Color tw4C2800 = Color(0XFF4C2800);
  static const Color tw2479D1 = Color(0XFF2479D1);
  static const Color twF2F2F2 = Color(0XFFF2F2F2);
  static const Color tw2478D2 = Color(0XFF2478D2);
  static const Color twC7D0D9 = Color(0XFFC7D0D9);
  static const Color twCCD7E2 = Color(0XFFCCD7E2);
  static const Color tw2C81CD = Color(0XFF2C81CD);
  static const Color tw1D60AF = Color(0XFF1D60AF);
  static const Color twFF8B08 = Color(0XFFFF8B08);
  static const Color twFFBE41 = Color(0XFFFFBE41);
  static const Color twE46824 = Color(0XFFE46824);
  static const Color tw085EC1 = Color(0XFF085EC1);
  static const Color twABABAB = Color(0XFFABABAB);
  static const Color tw660000 = Color(0XFF660000);
  static const Color twDBDBDB = Color(0XFFDBDBDB);
  static const Color tw000000 = Color(0XFF000000);
  static const Color tw99000000 = Color(0X99000000);
  static const Color tw1A000000 = Color(0X1A000000);
  static const Color tw00000000 = Color(0X00000000);
  static const Color tw666666 = Color(0XFF666666);
  static const Color twE4E4E6 = Color(0XFFE4E4E6);
  static const Color tw6B7072 = Color(0XFF6B7072);
  static const Color tw394043 = Color(0XFF394043);
  static const Color tw8B8B8B = Color(0XFF8B8B8B);
  static const Color twB1B1B1 = Color(0XFFB1B1B1);
  static const Color tw8C8C8C = Color(0XFF8C8C8C);
  static const Color twEBEBEB = Color(0XFFEBEBEB);
  static const Color tw333333 = Color(0XFF333333);
  static const Color twE0E0E0 = Color(0XFFE0E0E0);
  static const Color tw888888 = Color(0XFF888888);
  static const Color tw222222 = Color(0XFF222222);
  static const Color twFF4400 = Color(0XFFFF4400);
  static const Color twB2B2B2 = Color(0XFFB2B2B2);
  static const Color twB3B3B3 = Color(0XFFB3B3B3);
  static const Color tw2B65E7 = Color(0XFF2B65E7);
  static const Color tw4990FE = Color(0XFF4990FE);
  static const Color tw4D8DD9 = Color(0XFF4D8DD9);
  static const Color tw58E672 = Color(0XFF58E672);
  static const Color tw43C15B = Color(0XFF43C15B);
  static const Color twFB9999 = Color(0XFFFB9999);
  static const Color twF66B6B = Color(0XFFF66B6B);
  static const Color twFDB179 = Color(0XFFFDB179);
  static const Color twF0F0F0 = Color(0XFFF0F0F0);
  static const Color twFF9446 = Color(0XFFFF9446);
  static const Color tw4D4D4D = Color(0XFF4D4D4D);
  static const Color tw808080 = Color(0XFF808080);
  static const Color twF7F4CD = Color(0XFFF7F4CD);
  static const Color tw122A44 = Color(0XFF122A44);
  static const Color tw3C3C3C = Color(0XFF3C3C3C);
  static const Color twE6E6E6 = Color(0XFFE6E6E6);
  static const Color tw191919 = Color(0XFF191919);
  static const Color tw95ABC2 = Color(0XFF95ABC2);
  static const Color twF2F6FA = Color(0XFFF2F6FA);
  static const Color tw80F2F6FA = Color(0X80F2F6FA);
  static const Color twBFBFBF = Color(0XFFBFBFBF);
  static const Color twCCCCCC = Color(0XFFCCCCCC);
  static const Color twD9D9D9 = Color(0XFFD9D9D9);
  static const Color twE27000 = Color(0XFFE27000);
  static const Color twFFF7F0 = Color(0XFFFFF7F0);
  static const Color tw1E66B3 = Color(0XFF1E66B3);
  static const Color twC6DAF0 = Color(0XFFC6DAF0);
  static const Color twFAFAFA = Color(0XFFFAFAFA);
  static const Color twFFA800 = Color(0XFFFFA800);
  static const Color twFF7F00 = Color(0XFFFF7F00); //导航栏颜色
  static const Color twF6F6F6 = Color(0XFFF6F6F6);
  static const Color twFD9228 = Color(0XFFFD9228);
  static const Color twDDDDDD = Color(0XFFDDDDDD);
  static const Color tw1E61C8 = Color(0XFF1E61C8);
  static const Color twEEEEEE = Color(0XFFEEEEEE);
  static const Color twF1F5FA = Color(0XFFF1F5FA);
  static const Color twDCE1E7 = Color(0XFFDCE1E7);
  static const Color twDCE8F4 = Color(0XFFDCE8F4);
  static const Color twFB2F38 = Color(0XFFFB2F38);
  static const Color tw447DD8 = Color(0XFF447DD8);
  static const Color twADADAD = Color(0XFFADADAD);
  static const Color twE6E8EB = Color(0XFFE6E8EB);
  static const Color tw616161 = Color(0XFF616161);
  static const Color tw228D5A = Color(0XFF228D5A);
  static const Color twE99232 = Color(0XFFE99232);
  static const Color twD83A40 = Color(0XFFD83A40);
  static const Color twFF3327 = Color(0XFFFF3327);
  static const Color tw339D31 = Color(0XFF339D31);
  static const Color tw00B0F0 = Color(0XFF00B0F0);
  static const Color twFFC000 = Color(0XFFFFC000);
  static const Color twED7D31 = Color(0XFFED7D31);
  static const Color tw436BFF = Color(0XFF436BFF);
  static const Color tw1A1A1A = Color(0XFF1A1A1A);
  static const Color twFFFBF7 = Color(0XFFFFFBF7);
  static const Color tw28303A = Color(0XFF28303A);
  static const Color twEF9F65 = Color(0XFFEF9F65);
  static const Color twEB5645 = Color(0XFFEB5645);
  static const Color tw949494 = Color(0XFF949494);
  static const Color tw32353A = Color(0XFF32353A);
  static const Color tw4E7DFF = Color(0XFF4E7DFF);
  static const Color twFFA244 = Color(0XFFFFA244);
  static const Color tw48C360 = Color(0XFF48C360);
  static const Color twEDEDED = Color(0XFFEDEDED);
  static const Color tw474747 = Color(0XFF474747);
  static const Color tw1E66B2 = Color(0XFF1E66B2);
  static const Color tw5A6276 = Color(0XFF5A6276);
  static const Color twFF2D2D = Color(0XFFFF2D2D);
  static const Color tw595959 = Color(0XFF595959);
  static const Color twF3F6FA = Color(0XFFF3F6FA);
  static const Color tw010101 = Color(0XFF010101);
  static const Color twF2F4F7 = Color(0XFFF2F4F7);
  static const Color tw262626 = Color(0XFF262626);
  static const Color twC7C7C7 = Color(0XFFC7C7C7);
  static const Color tw737373 = Color(0XFF737373);
  static const Color twE2E2E2 = Color(0XFFE2E2E2);
  static const Color twC88B18 = Color(0XFFC88B18);
  static const Color twFEFCEE = Color(0XFFFEFCEE);
  static const Color twF3D995 = Color(0XFFF3D995);
  static const Color twE1E1E1 = Color(0XFFE1E1E1);
  static const Color tw7A7A7A = Color(0XFF7A7A7A);
  static const Color tw777777 = Color(0XFF777777);
  static const Color tw444444 = Color(0XFF444444);
  static const Color twE32D2D = Color(0XFFE32D2D);
  static const Color twC2C2C2 = Color(0XFFC2C2C2);
  static const Color twAAAAAA = Color(0XFFAAAAAA);
  static const Color twFF6A10 = Color(0XFFFF6A10);
  static const Color twF6BE85 = Color(0XFFF6BE85);
  static const Color twFFE2C4 = Color(0XFFFFE2C4);
  static const Color twE37047 = Color(0XFFE37047);
  static const Color twF9F9F9 = Color(0XFFF9F9F9);
  static const Color tw060606 = Color(0XFF060606);
  static const Color tw196BBF = Color(0XFF196BBF);
  static const Color twACACAC = Color(0XFFACACAC);
  static const Color twF3BF85 = Color(0XFFF3BF85);
  static const Color twF7F7F7 = Color(0XFFF7F7F7);
  static const Color tw293355 = Color(0XFF293355);
  static const Color tw292929 = Color(0XFF292929);
  static const Color twFF5B01 = Color(0XFFFF5B01);
  static const Color tw7E9CE1 = Color(0XFF7E9CE1);
  static const Color tw61C0B7 = Color(0XFF61C0B7);
  static const Color twECAC72 = Color(0XFFECAC72);
  static const Color twFC3A1A = Color(0XFFFC3A1A);
  static const Color twFFB56B = Color(0XFFFFB56B);
  static const Color twFFDEBC = Color(0XFFFFDEBC);
  static const Color twF2A253 = Color(0XFFF2A253);
  static const Color twF4B777 = Color(0XFFF4B777);
  static const Color tw2A2A2A = Color(0XFF2A2A2A);
  static const Color twFF5A4E = Color(0XFFFF5A4E);
  static const Color tw0044DD = Color(0XFF0044DD);
  static const Color tw8594E5 = Color(0XFF8594E5);
  static const Color tw62B461 = Color(0XFF62B461);
  static const Color twF2AB41 = Color(0XFFF2AB41);
  static const Color tw9AABBE = Color(0XFF9AABBE);
  static const Color twD5D5D5 = Color(0XFFD5D5D5);
  static const Color tw507DAF = Color(0XFF507DAF);
  static const Color twEE7C78 = Color(0XFFEE7C78);
  static const Color tw4B7183 = Color(0XFF4B7183);
  static const Color twFCECD7 = Color(0XFFFCECD7);
  static const Color tw1D72C9 = Color(0XFF1D72C9);
  static const Color twE16600 = Color(0XFFE16600);
  static const Color tw4FC781 = Color(0XFF4FC781);
  static const Color tw4E71A5 = Color(0XFF4E71A5); //A0B4C8  F2F6FA
  static const Color twA0B4C8 = Color(0XFFA0B4C8);
  static const Color tw9FB4C8 = Color(0XFF9FB4C8);
  static const Color tw339FB4C8 = Color(0X339FB4C8);
  static const Color tw555555 = Color(0XFF555555);
  static const Color tw1F870C = Color(0XFF1F870C);
  static const Color twF74F01 = Color(0XFFF74F01);
  static const Color tw4A4A4A = Color(0XFF4A4A4A);
  static const Color twFFF9D0 = Color(0XFFFFF9D0);
  static const Color twFFF9F1 = Color(0XFFFFF9F1);
  static const Color twFCFCFC = Color(0XFFFCFCFC);
  static const Color twF8FAFC = Color(0XFFF8FAFC);
  static const Color twECB461 = Color(0XFFECB461);
  static const Color tw1378E5 = Color(0XFF1378E5);
  static const Color tw42CC8F = Color(0XFF42CC8F); //预推案
  static const Color tw599DFF = Color(0XFF599DFF);
  static const Color tw979797 = Color(0XFF979797);
  static const Color twF66048 = Color(0XFFF66048);
  static const Color twFF5B20 = Color(0XFFFF5B20);
  static const Color twE8E8E8 = Color(0XFFE8E8E8);
  static const Color twFFBC75 = Color(0XFFFFBC75);
  static const Color twFAF5F2 = Color(0XFFFAF5F2);
  static const Color twFFE500 = Color(0XFFFFE500);
  static const Color tw8D3D10 = Color(0XFF8D3D10);
  static const Color tw0274E5 = Color(0XFF0274E5);
  static const Color tw63461A = Color(0XFF63461A);
  static const Color tw654922 = Color(0XFF654922);
  static const Color twDFBB80 = Color(0XFFDFBB80);
  static const Color twD86C00 = Color(0XFFD86C00);
  static const Color tw1C61AE = Color(0XFF1C61AE);
  static const Color twFEF4E3 = Color(0XFFFEF4E3);
  static const Color tw3999F0 = Color(0XFF3999F0);
  static const Color tw9ECDFF = Color(0XFF9ECDFF);
  static const Color tw121212 = Color(0XFF121212); // // 暗黑模式下的导航栏黑色
  static const Color tw1C1C1C = Color(0XFF1C1C1C);
  static const Color tw1C1C1E = Color(0XFF1C1C1E);
  static const Color tw6E6E6E = Color(0XFF6E6E6E);
  static const Color tw272729 = Color(0XFF272729);
  static const Color twE9F3FB = Color(0XFFE9F3FB);
  static const Color tw83BFFF = Color(0XFF83BFFF);
  static const Color twF5FBFF = Color(0XFFF5FBFF);
  static const Color tw181818 = Color(0XFF181818);
  static const Color tw505050 = Color(0XFF505050);
  static const Color twFFF7E6 = Color(0XFFFFF7E6);
  static const Color tw126AFF = Color(0XFF126AFF);
  static const Color twF0F6FF = Color(0XFFF0F6FF);
  static const Color twF5F7F9 = Color(0xFFF5F7F9);
  static const Color twECF3FF = Color(0xFFECF3FF);
  static const Color tw9B5B1A = Color(0xFF9B5B1A);
  static const Color tw67CDDB = Color(0xFF67CDDB);
  static const Color tw3367CDDB = Color(0x3367CDDB);
  static const Color tw09B47C = Color(0xFF09B47C);
  static const Color twE7F3FF = Color(0xFFE7F3FF); // twF3F7FC
  static const Color twF3F7FC = Color(0xFFF3F7FC); // tw80AFFF
  static const Color tw80AFFF = Color(0xFF80AFFF); // tw80AFFF
  static const Color twEAF0F8 = Color(0xFFEAF0F8);
  static const Color twFFF3E6 = Color(0xFFFFF3E6);
  static const Color tw67461F = Color(0xFF67461F);
  static const Color twFC7616 = Color(0xFFFC7616);
  static const Color tw358FFF = Color(0xFF358FFF);
  static const Color tw3587FF = Color(0xFF3587FF); // Test
  static const Color twC4DAFF = Color(0xFFC4DAFF);
  static const Color twCDE0FF = Color(0xFFCDE0FF);
  static const Color twA7C8FF = Color(0xFFA7C8FF);
  static const Color twF4F8FF = Color(0xFFF4F8FF);
  static const Color twF1F6FF = Color(0xFFF1F6FF);
  static const Color twFEDDC3 = Color(0xFFFEDDC3);
  static const Color twFFBF99 = Color(0xFFFFBF99);
  static const Color twB15F2D = Color(0xFFB15F2D);
  static const Color twFFEEDE = Color(0xFFFFEEDE);
  static const Color twF65548 = Color(0xFFF65548);
  static const Color tw63461a = Color(0xFF63461a);

  static const Color twFF7F02 = Color(0xFFFF7F02);
  static const Color twF65547 = Color(0xFFF65547);
  static const Color twFF2A00 = Color(0xFFFF2A00);
  static const Color tw14FF8000 = Color(0x14FF8000);
  static const Color tw00FF0000 = Color(0x00FF0000);
  static const Color twE60012 = Color(0xFFE60012);
  static const Color tw565656 = Color(0xFF565656);
  static const Color twFFE9CC = Color(0xFFFFE9CC);
  static const Color twF2F7FF = Color(0xFFF2F7FF);
  static const Color twFFF9F2 = Color(0xFFFFF9F2);
  static const Color tw597DB3 = Color(0xFF597DB3);
  static const Color tw00CC22 = Color(0xFF00CC22);
  static const Color twFAAC15 = Color(0xFFFAAC15);
  static const Color twF4F5F6 = Color(0XFFF4F5F6);
  static const Color twFF9A33 = Color(0XFFFF9A33);
  static const Color twf1f4f7 = Color(0XFFf1f4f7);
  static const Color twEFFAFB = Color(0XFFEFFAFB);
  static const Color twF06800 = Color(0XFFF06800);
  static const Color tw264780 = Color(0XFF264780);
  static const Color tw8094A7 = Color(0xFF8094A7);

  static const Color twB36536 = Color(0xFFB36536);
  static const Color twA6594F = Color(0xFFA6594F);
  static const Color tw9D6643 = Color(0xFF9D6643);
  static const Color twB37047 = Color(0xFFB37047);
  static const Color tw8F502B = Color(0xFF8F502B);
  static const Color twB34840 = Color(0xFFB34840);
  static const Color tw99433C = Color(0xFF99433C);
  static const Color tw865DA0 = Color(0xFF865DA0);
  static const Color tw965F2F = Color(0xFF965F2F);
  static const Color twC34444 = Color(0xFFC34444);
  static const Color tw66367D = Color(0xFF66367D);
  static const Color tw9E6742 = Color(0xFF9E6742);
  static const Color twA6584F = Color(0xFFA6584F);
  static const Color tw975F2F = Color(0xFF975F2F);
  static const Color twA63B3B = Color(0xFFA63B3B);
  static const Color tw693486 = Color(0xFF693486);

  static const Color tw337EFF = Color(0xFF337EFF);
  static const Color twFFB366 = Color(0xFFFFB366);
  static const Color tw0048F0 = Color(0xFF0048F0);
  static const Color twFE4704 = Color(0xFFFE4704);
  static const Color twFEA309 = Color(0xFFFEA309);
  static const Color tw669EFF = Color(0xFF669EFF);
  static const Color twE1EDFF = Color(0xFFE1EDFF);
  static const Color tw6EA4FF = Color(0xFF6EA4FF);
  static const Color twF7F8FA = Color(0xFFF7F8FA);
  static const Color twFFE6CC = Color(0xFFFFE6CC);

  static const Color tw00A670 = Color(0xFF00A670);
  static const Color tw00A46F = Color(0xFF00A46F);
  static const Color tw7A9BCC = Color(0xFF7A9BCC);
  static const Color twFFE1B5 = Color(0xFFFFE1B5);

  static const Color twF2FFFB = Color(0xFFF2FFFB);
  static const Color tw47B3A1 = Color(0xFF47B3A1);
  static const Color twC2D5F2 = Color(0xffC2D5F2);
  static const Color twF0F4FA = Color(0xffF0F4FA);

  static const Color twFFF5CC = Color(0xffFFF5CC);
  static const Color twF01800 = Color(0xffF01800);
  static const Color twCE9953 = Color(0xffCE9953);
  static const Color twFFF6EB = Color(0xffFFF6EB);
  static const Color twFF9326 = Color(0xffFF9326);
  static const Color twFFCC99 = Color(0xffFFCC99);
}


example/lib/main.dart
/*
 * @Author: zhengzeqin
 * @Date: 2022-07-17 22:49:40
 * @LastEditTime: 2022-08-03 16:14:10
 * @Description: your project
 */
import 'package:flutter/material.dart';
import 'package:flutter_keyboard_visibility/flutter_keyboard_visibility.dart';
import 'package:tw_step_counter/tw_colors.dart';
import 'package:tw_step_counter/tw_step_counter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final controller = TWStepCounterController();
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: KeyboardDismissOnTap(
        child: Scaffold(
          appBar: AppBar(
            title: const Text('example app'),
          ),
          body: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                TextButton(
                  child: const Text('失去焦点'),
                  onPressed: () {
                    controller.unFocus();
                  },
                ),
                TextButton(
                  child: const Text('获取焦点'),
                  onPressed: () {
                    controller.getFocus();
                  },
                ),
                TextButton(
                  child: const Text('是否输入'),
                  onPressed: () {
                    print('最后是否输入 ${controller.isInputModify}');
                  },
                ),
                TextButton(
                  child: const Text('更新值为 100'),
                  onPressed: () {
                    controller.modifyValue(100);
                  },
                ),
                SizedBox(
                  width: 300,
                  child: TWStepCounter(
                    unit: '元/天',
                    currentValue: 130,
                    mixValue: 75,
                    maxValue: 250,
                    onTap: (value) {
                      print('点击回调===>$value');
                    },
                    inputTap: (value) {
                      print('输入回调===>$value');
                    },
                    controller: controller,
                    addForbiddenIcon: 'assets/forbidden_add.png',
                    addIcon: 'assets/add.png',
                    reduceForbiddenIcon: 'assets/forbidden_reduce.png',
                    reduceIcon: 'assets/reduce.png',
                    defaultColor: TWColors.twF5F5F5,
                    highlightColor: Colors.yellow,
                    borderLineColor: Colors.red,
                    inputMultipleValue: 5,
                    // limitLength: 8,
                    // height: 70,
                    // isUpdateInLimitValue: false,
                    // isSupportAnimation: false,
                    decimalsCount: 2,
                    btnWidth: 64,
                    height: 30,
                    isUpdateInputChange: false,
                    valuePadding: const EdgeInsets.only(
                      left: 10,
                      right: 10,
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
